### PR TITLE
recorder: fix a couple of memory leaks

### PR DESCRIPTION
**Pull request by Dudemanguy**
Not sure how long these have been around but it leaked on every packet.

### CODE REVIEW

File: common/recorder.c [hunk 1]
```diff
@@ -254,6 +261,8 @@ static void mux_packet(struct mp_recorder_sink *rst,
 
     if (av_interleaved_write_frame(priv->mux, new_packet) < 0)
         MP_ERR(priv, "Failed writing packet.\n");
+
+    av_packet_free(&new_packet);
```
Comment (sfan5) [inline common/recorder.c:265]: [docs](https://ffmpeg.org/doxygen/5.1/group__lavf__encoding.html#ga37352ed2c63493c38219d935e71db6c1) of av_interleaved_write_frame:

> If the packet is reference-counted, this function will take ownership of this reference and unreference it later when it sees fit. If the packet is not reference-counted, libavformat will make a copy. The returned packet will be blank (as if returned from, even on error.

so this isn't actually needed

Comment (Traneptora) [reply common/recorder.c:265]: Is the packet reference counted though? If not, it'll be copied and we still need to free it.

Comment (Dudemanguy) [reply common/recorder.c:265]: ASAN complains without this so I just assumed it wasn't reference-counted and was indeed copied.

Comment (Dudemanguy) [reply common/recorder.c:265]: I think so. I'm not totally sure what the intent is supposed to be just glancing at [`interleaved_write_packet`](https://github.com/FFmpeg/FFmpeg/blob/3951c1899ece9c968587d8a0f3e9ab7498669117/libavformat/mux.c#L1249).

```
static int interleaved_write_packet(AVFormatContext *s, AVPacket *pkt,
                                    int flush, int has_packet)
{
    FFFormatContext *const si = ffformatcontext(s);
    for (;; ) {
        int ret = si->interleave_packet(s, pkt, flush, has_packet);
        if (ret <= 0)
            return ret;

        has_packet = 0;

        ret = write_packet(s, pkt);
        av_packet_unref(pkt);
        if (ret < 0)
            return ret;
    }
}
```

It does appear to free the packet in some cases, but not if the ret is <= 0 which I would assume happens quite often (I presume 0 is not an error).

**Pull request merged**